builtin/credential/okta/backend.go

```diff
@@ -241,6 +241,8 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username, pas
 		if rsp == nil {
 			return nil, logical.ErrorResponse("okta auth backend unexpected failure"), nil, nil
 		}
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
```

Comment:
This comment applies to all the changes in this PR, not just this one.

I believe this code introduces subtly different behavior than we had before, and I'm not certain if it's problematic or not. Reading the docs for `time.After(d Duration)`, it says that it waits for the duration `d` to elapse, then sends the current time on the returned channel. We're not using the returned time, so that part is irrelevant. But it does sound like it's a single fire mechanism. The docs mention that it's equivalent to `NewTimer(d).C`.

I mention it because `NewTicker()` is different from `NewTimer()` in that `NewTimer()` creates a timer that fires once, like `time.After()` but `NewTicker(d Duration)` creates a ticker that fires repeatedly, every `d` interval. My possibly misplaced and/or invalid concern is would the repeated firing of the ticker cause other channels in the `select` statement to not be evaluated as soon as they normally would?

Mostly I'm just concerned about the change in behavior, even though it might be innocuous. How would you feel about changing `NewTicker()` to `NewTimer()`, since it sounds like those are equivalent?

Reply:
Not just equivalent I'm afraid:

```
func After(d Duration) <-chan Time {
	return NewTimer(d).C
}
```

Reply:
I think we can use NewTimer, but we have to ensure that when we take other select branches than the one where the timer channel fires, we must 

```
if !t.Stop() {
     <-t.C
}
```

Reply:
I'm also ok skipping my suggestion completely and sticking with `NewTicker()`. I'm not sure if my concern is well founded or likely to actually occur (my sense is probably not likely).

Reply:
> repeated firing of the ticker cause other channels in the select statement to not be evaluated as soon as they normally would?

true, if there's a buildup of ticks that could happen. I can switch this to a timer, and try to add a semgrep rule to catch cases where the timer channel isn't closed in the other select cases.

Reply:
switched to using timers in all locations, but i can't seem to find a way to write the semgrep rule to detect the timer not being stopped.